drivers/hud_spm_report/app/models/hud_spm_report/fy2023/episode.rb

```diff
@@ -102,11 +81,17 @@ def compute_episode(enrollments, included_project_types:, excluded_project_types
         if enrollment.project_type.in?(HudUtility2024.project_type_number_from_code(:es_nbn))
           # NbN only gets service nights in the report range
           bed_nights.merge!(
-            enrollment.enrollment.
-              services.merge(GrdaWarehouse::Hud::Service.bed_night.between(start_date: report_start_date, end_date: report_end_date)).
-              pluck(s_t[:id], s_t[:date_provided]).map do |service_id, date|
+            enrollment.enrollment.services. # Preloaded
+              # merge(GrdaWarehouse::Hud::Service.bed_night.between(start_date: report_start_date, end_date: report_end_date)).
+              # pluck(s_t[:id], s_t[:record_type], s_t[:date_provided]).
+              map do |service|
+                service_id = service.id
+                record_type = service.record_type
+                date = service.date_provided
+                next unless record_type == HudUtility2024.record_type('Bed Night', true) && date.between?(report_start_date, report_end_date)
+
                 [enrollment, service_id, date]
```

Comment:
maybe: 
```suggestion
              map do |service|
                bed_night = service.record_type == == HudUtility2024.record_type('Bed Night', true)
                date = service.date_provided
                next unless bed_night && date.between?(report_start_date, report_end_date)

                [enrollment, service.id, date]
```

Since we aren't using service_id more than once, and the only reason to assign the record type is to determine if it is a bedight, seems like it's a bit cleaner to only assign the boolean, and then just pull the id from the record later.

Reply:
The local variables were left to keep the code as close to the `pluck` based forms as possible in case we wanted it back. But, there does seem to be a significant performance speed up from these changes -- so I cleaned this up